meos/include/meos_internal.h

```diff
@@ -287,6 +287,49 @@ extern void stbox_set_box3d(const STBox *box, BOX3D *box3d);
  * Functions for temporal types
  *****************************************************************************/
 
+/* Macros for speeding up access to components of temporal sequences (sets)*/
+
+#ifdef DEBUG_BUILD
+extern const TInstant *tsequence_inst_n(const TSequence *seq, int index);
+extern const TSequence *tsequenceset_seq_n(const TSequenceSet *ss, int index);
+#else
+/**
+ * @brief Return a pointer to the offsets array of a temporal sequence
+ * @note The period component of the bbox is already declared in the struct
+ */
+#define tsequence_offsets_ptr(seq) ( (size_t *)( \
+  ((char *) &((seq)->period)) + (seq)->bboxsize ) )
+
+/**
+ * @brief Return the n-th instant of a temporal sequence.
+ * @note The period component of the bbox is already declared in the struct
+ * @pre The argument @p index is less than the number of instants in the
+ * sequence
+ */
+#define tsequence_inst_n(seq, index) ( (TInstant *)( \
+  ((char *) &((seq)->period)) + (seq)->bboxsize + \
+  (sizeof(size_t) * (seq)->maxcount) + (tsequence_offsets_ptr(seq))[index] ) )
+
+/**
+ * @brief Return a pointer to the offsets array of a temporal sequence set
+ * @note The period component of the bbox is already declared in the struct
+ */
+#define tsequenceset_offsets_ptr(ss) ( (size_t *)( \
+  ((char *) &((ss)->period)) + (ss)->bboxsize ) )
+
+/**
+ * @brief Return the n-th sequence of a temporal sequence set
+ * @note The period component of the bbox is already declared in the struct
+ * @pre The argument @p index is less than the number of sequences in the
+ * sequence set
+ */
+#define tsequenceset_seq_n(ss, index) ( (TSequence *)( \
+  ((char *) &((ss)->period)) + (ss)->bboxsize + \
+  (sizeof(size_t) * (ss)->maxcount) + (tsequenceset_offsets_ptr(ss))[index] ) )
```

Comment:
Should we write these function using CAPS to denote the fact that they are macros?
Just so we know to avoid writing something like:
`tsequence_inst_n(long_function_returning_a_seq(...), i)`
since this would cause the long function to be evaluated multiple times.